### ISSUE TITLE
json-file logging driver is implicitly set when tag is specified

### ISSUE THREAD

**Issue by bobrik** [docker/docker#20034]
I'm using docker 1.10 with #19564 applied on top. systemd unit file has some extra options:

```
ExecStart=/usr/bin/docker daemon -H fd:// --storage-driver overlay --log-driver journald --log-opt tag=
```

Running docker containers without any logging options works just fine:

```
$ docker run --rm -it --net host --name whoa alpine:3.3 echo hell yeah
hell yeah
```

While overwriting tag also sets logging driver to `json-file`:

```
$ docker run --rm -it --log-opt 'tag=ffuuu.{{.ID}}' --net host --name whoa alpine:3.3 echo hell yeah
docker: Error response from daemon: unknown log opt 'tag' for json-file log driver.
```

**Comment by bobrik** [comment]
This seems to be fixed in 1.12.0

**Comment by harobed** [comment]
```
$  docker version
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 17:32:24 2016
 OS/Arch:      darwin/amd64
 Experimental: true

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 17:32:24 2016
 OS/Arch:      linux/amd64
 Experimental: true

$  docker run --log-opt "tag={{.ImageName}}/{{.Name}}/{{.ID}}" --rm -it ubuntu bash
docker: Error response from daemon: unknown log opt 'tag' for json-file log driver.
See 'docker run --help'.
 ```
